dataprofiler/profilers/column_profile_compilers.py

```diff
@@ -15,20 +15,20 @@ class BaseColumnProfileCompiler(with_metaclass(abc.ABCMeta, object)):
     # NOTE: these profilers are ordered. Test functionality if changed.
     _profilers = list()
 
-    def __init__(self, df_series, options=None):
+    def __init__(self, df_series, options=None, data_labeler=None):
         if not self._profilers:
             raise NotImplementedError("Must add profilers.")
 
         self.name = df_series.name
         self._profiles = OrderedDict()
-        self._create_profile(df_series, options)
+        self._create_profile(df_series, options, data_labeler)
 
     @property
     @abc.abstractmethod
     def profile(self):
         raise NotImplementedError()
 
-    def _create_profile(self, df_series, options=None):
```

Comment:
If possible we should think of another option to pass the data labeler instead of changing the functionality for all.

Potentially if it gets sent in options and passed to all.